Sources/ArgumentParser/Completions/ZshCompletionsGenerator.swift

```diff
@@ -12,12 +12,13 @@
 struct ZshCompletionsGenerator {
   /// Generates a Zsh completion script for the given command.
   static func generateCompletionScript(_ type: ParsableCommand.Type) -> String {
-    let initialFunctionName = [type].completionFunctionName()
+    let initialFunctionName = [type].completionFunctionName().zshEscapingCommandName()
+    let commandName = type._commandName.zshEscapingCommandName()
 
     return """
-    #compdef \(type._commandName)
+    #compdef \(commandName)
```

Comment:
I don't think this part can change — IIUC the `#compdef` tells Zsh what command the completions are for, so they need to match the executable.